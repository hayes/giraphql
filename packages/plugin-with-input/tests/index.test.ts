import { gql } from 'apollo-server-core';
import { execute, printSchema } from 'graphql';
import schema from './example/schema';

describe('example schema', () => {
  it('generates expected schema', () => {
    expect(printSchema(schema)).toMatchSnapshot();
  });

  describe('query', () => {
    it('returns expected result', async () => {
      const query = gql`
        query {
          exampleQuery(input: { id: "123" }) {
            id
          }
          withOptions(custom: { id: "123" }) {
            id
          }
          obj {
            exampleObjectField(input: { id: "123" }) {
              id
            }
            withOptions(custom: { id: "123" }) {
              id
            }
          }
          iface {
            __typename
            exampleInterfaceField(input: { id: "123" }) {
              id
            }
            withOptions(custom: { id: "123" }) {
              id
            }
          }
        }
      `;

      const result = await execute({
        schema,
        document: query,
        contextValue: {},
      });

      expect(result).toMatchInlineSnapshot(`
        Object {
          "data": Object {
            "exampleQuery": "123",
            "iface": Object {
              "__typename": "ObjWithInterface",
              "exampleInterfaceField": "123",
              "withOptions": "123",
            },
            "obj": Object {
              "exampleObjectField": "123",
              "withOptions": "123",
            },
            "withOptions": "123",
          },
        }
      `);
    });
  });

  describe('mutations', () => {
    it('returns expected result', async () => {
      const query = gql`
        mutation {
          exampleMutation(input: { id: "123" }) {
            id
          }
          withOptions(custom: { id: "123" }) {
            id
          }
        }
      `;

      const result = await execute({
        schema,
        document: query,
        contextValue: {},
      });

      expect(result).toMatchInlineSnapshot(`
        Object {
          "data": Object {
            "exampleMutation": "123",
            "withOptions": "123",
          },
        }
      `);
    });
  });
});
