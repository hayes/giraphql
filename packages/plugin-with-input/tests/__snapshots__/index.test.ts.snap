// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`example schema generates expected schema 1`] = `
"input ExampleInterfaceFieldInput {
  id: ID!
}

input ExampleMutationInput {
  id: ID!
}

input ExampleObjectFieldInput {
  id: ID!
}

input ExampleQueryInput {
  id: ID!
}

input ExampleSubscriptionInput {
  id: ID!
}

type Mutation {
  exampleMutation(
    \\"\\"\\"input arg\\"\\"\\"
    input: ExampleMutationInput!
  ): ID!

  \\"\\"\\"mutation field\\"\\"\\"
  withOptions(
    \\"\\"\\"input arg\\"\\"\\"
    custom: TestMutationInput!
  ): ID!
}

type ObjWithInterface implements TestInterface {
  exampleInterfaceField(
    \\"\\"\\"input arg\\"\\"\\"
    input: ExampleInterfaceFieldInput!
  ): ID!
  id: ID!

  \\"\\"\\"interface field\\"\\"\\"
  withOptions(
    \\"\\"\\"input arg\\"\\"\\"
    custom: TestInterfaceInput!
  ): ID!
}

type Query {
  exampleQuery(
    \\"\\"\\"input arg\\"\\"\\"
    input: ExampleQueryInput!
  ): ID!
  iface: TestInterface!
  obj: TestObj!

  \\"\\"\\"query field\\"\\"\\"
  withOptions(
    \\"\\"\\"input arg\\"\\"\\"
    custom: TestQueryInput!
  ): ID!
}

type Subscription {
  exampleSubscription(
    \\"\\"\\"input arg\\"\\"\\"
    input: ExampleSubscriptionInput!
  ): ID!

  \\"\\"\\"subscription field\\"\\"\\"
  withOptions(
    \\"\\"\\"input arg\\"\\"\\"
    custom: TestSubscriptionInput!
  ): ID!
}

interface TestInterface {
  exampleInterfaceField(
    \\"\\"\\"input arg\\"\\"\\"
    input: ExampleInterfaceFieldInput!
  ): ID!
  id: ID!

  \\"\\"\\"interface field\\"\\"\\"
  withOptions(
    \\"\\"\\"input arg\\"\\"\\"
    custom: TestInterfaceInput!
  ): ID!
}

\\"\\"\\"input field\\"\\"\\"
input TestInterfaceInput {
  id: ID!
}

\\"\\"\\"input field\\"\\"\\"
input TestMutationInput {
  id: ID!
}

type TestObj {
  exampleObjectField(
    \\"\\"\\"input arg\\"\\"\\"
    input: ExampleObjectFieldInput!
  ): ID!
  id: ID!

  \\"\\"\\"object field\\"\\"\\"
  withOptions(
    \\"\\"\\"input arg\\"\\"\\"
    custom: TestObjectInput!
  ): ID!
}

\\"\\"\\"input field\\"\\"\\"
input TestObjectInput {
  id: ID!
}

\\"\\"\\"input field\\"\\"\\"
input TestQueryInput {
  id: ID!
}

\\"\\"\\"input field\\"\\"\\"
input TestSubscriptionInput {
  id: ID!
}
"
`;
